### DISCUSSION TITLE
How is the structure+expectations so far?

### DISCUSSION THREAD

**Discussion by cooperrc** [cooperrc/computational-mechanics#40] [Ideas]
Greetings,
Excellent work in this first 1.5 weeks :+1: 

The first thing I asked you to do was to discuss changes you would suggest to help you succeed in the course. I know time was heavily requested, but that part I can't really change. 

You have turned in one HW + one project. How did you get help? What worked? What was difficult?

Is there any part of the structure/grading/communication that we should change in the syllabus for the next modules? Post your ideas/questions/reflections here to join the discussion.

**Comment by Aidan-Myatt** [comment]
From what I have seen so far, the homework has a lot more problems and is slightly more difficult than the projects. Would it be possible to make some of the problems on the homework a bonus or something similar, especially considering the timeframe? I am not sure how different the future homework and projects will be. This is just my first impression.

**Reply by FishMacaron** [reply]
I agree with the sentiment about the homework, I found it hard to manage my time with the first homework solely due to the sheer amount of problems. I try to take on one part a day at least but sometimes fall behind when I end up having to spend a long period of time on a question or part of a question I have a lot of difficulty with. As of right now, I feel fairly on track for homework 2 though.

**Comment by JordanAlnaimi** [comment]
I do feel like HW is need a lot more time to work on. For me the video professor posted about the HW and the project helped me. It will be good if we have more video about HW and project that help us start.

**Reply by UniquelyAbbi** [reply]
I agree with this, im not good at coding and this class is my first major coding. Videos would be very helpful

**Comment by y1bo44** [comment]
I thougth the HW1 and project1 are little hard for me as a new code learner and professor's video do helped.I hope professor could provide some more references code or illustrated video.That would help hugely.

**Comment by VictoireHA** [comment]
The problems in HW1 are a bit obscure to understand. I struggled to comprehend the statements in the problems.

**Comment by caroline-cook** [comment]
I also found the homework to be harder than the project. However, I find that the tutorial/reference videos that walk us through certain code makes it much easier to understand. For me at least, because I'm a visual learner, learning code techniques through reading a module page is more of a challenge than seeing it being done. Other than that the material has been straightforward and not too demanding!

**Reply by JordanAlnaimi** [reply]
I agree, I feel like watching a video will help me more then reading, at least for me.

**Reply by cooperrc** [reply]
:+1: I'll keep adding videos in the Classroom page then

**Comment by feliciones** [comment]
I am in agreeance with what a number of other students are saying, the hw was more difficult than the project, and the video was extremely useful in solving the hw. It would also be nice to see more collaboration in the discussion threads so that we can all work together and help eachother.

**Reply by JordanAlnaimi** [reply]
It will be nice if someone create some kind of form we can contact each other.

**Comment by cooperrc** [comment]
> From what I have seen so far, the homework has a lot more problems and is slightly more difficult than the projects. Would it be possible to make some of the problems on the homework a bonus or something similar, especially considering the timeframe? 

Unfortunately, there are no shortcuts here. The HW will definitely be longer than the project. For this current module_02, the rough breakdown should be:
- 1 hr/reading = 4 hrs *if the HW was difficult, try taking notes and doing the exercises*
- 2 hr/HW part = 8 hrs
- 4-6 hr/project = 4-6 hrs
- **total = 16-18 hrs**

We are learning some fundamental programming concepts, but we need some new skills to develop in order to implement these ideas. The only way to build a new skill is practice. That's why you can resubmit any assignment. You can attempt the problem and write in questions or show code with errors in it and I'll give you guidance to resubmit.

**Comment by taa18009** [comment]
I think my main feedback would be that while doing the homework it gets confusing to follow because I feel we are jumping back and forth between data sets. I think rather than splitting the homework about into problem parts it would be easier to work with one data set completely, then move on to the next. Rather than doing a couple things with one, then moving on to another, then going back to work on the first one again.

**Comment by C41ynn1** [comment]
So far, I feel like the homework has been the hardest part for me. The project felt more manageable, but the homework took more time because some of the questions were hard to fully understand at first.
The videos helped me the most because I learn better when I can see the steps being done instead of only reading directions. I also used the class notes and examples to help figure out how to start the code. The hardest part was knowing exactly what each problem wanted and making sure I was using the right variables.
For future modules, I think it would help to have more short videos or examples that show how to begin the homework problems. It would also be helpful if the instructions were a little clearer, especially for students who are still new to coding.

**Comment by amoduconn** [comment]
I think having videos for all the homeworks will be better. Labelled videos on the Youtube page will be beneficial.

**Comment by sierra-garofalo** [comment]
I see this was posted awhile ago, but I do have some input and would like to share. I definitely agree that the YouTube videos were helpful, and that it would be nice if they were labeled and organized accordingly so we can find them easier. Also, if there was a way to condense them a little bit so that they are more likely to be finished in one sitting, it would be easier to keep up with the material. Since the homeworks take a very long time to complete, it would be nice to spend less time with specifically-timed supplemental material and be able to interact with self-paced examples on Jupyter. Some other smaller supplemental materials, like specific examples in shorter (less likely to be interrupted) videos would also be helpful.

I also saw in another comment that you posted that the projects should take longer, but the homework was honestly the longest, hardest part. It felt like being thrown into it without much preparation. However after getting through that, they were great preparation for the projects. If there's a better way to bridge that gap between the readings and the homework, then I think more people would feel the homework is reasonable.

I know this might sound kind of simplistic or "hand-hold-y", but I think it's extremely necessary with learning to code with a language we likely haven't used in awhile: giving an example, and then having us recreate it by hand with only a _slight_ change in context. It's hard to find sufficient, money-free, AI-free (and therefore more accurate) resources on coding for specific problems. It takes longer to find these resources than to actually work on the assignments. If you would be willing to set up these kind of exercises (similar to the degree in chapter 1, but slightly more challenging maybe?), I think most students would take advantage of them and feel better prepared! A lot of the examples in the readings felt related in topic, but not execution, which is the most important part of this course.

Thank you for all your help and I hope you have a great rest of your summer!

**Reply by cooperrc** [reply]
Thanks for the feedback @sierra-garofalo. 

I think its about time to revise structure+expectations for the course. It was born in the Covid era and has been settling into a rhythm in the AI-everywhere era. 

I'm thinking HW might need to be revised into shorter, conceptual assignments while projects can focus more on the coding aspect. Thanks for joining this Summer!
